modules/external-api/src/main/java/org/opencastproject/external/endpoint/SeriesEndpoint.java

```diff
@@ -344,15 +359,45 @@ public JValue apply(SearchResultItem<Series> a) {
     }
   }
 
+  /**
+   * Get an {@link AccessControlList} from a {@link Series}.
+   *
+   * @param series
+   *          The {@link Series} to get the ACL from.
+   * @return The {@link AccessControlList} stored in the {@link Series}
+   */
+  private static AccessControlList getAclFromSeries(Series series) {
+    AccessControlList activeAcl = new AccessControlList();
+    try {
+      if (series.getAccessPolicy() != null)
+        activeAcl = AccessControlParser.parseAcl(series.getAccessPolicy());
+    } catch (Exception e) {
+      logger.error("Unable to parse access policy", e);
+    }
+    return activeAcl;
```

Comment:
Is "log error and continue with dummy value" a valid error handling style in Opencast? I would assume the API should respond with 5xx if parsing the ACL fails. But I am not sure, as I have no experience on how the API code usually handles these things.

Reply:
This is a tricky one. In general, this is probably not a good idea. However, if asking for a list of a series fails with 500 only because one ACL is broken, that's also gonna suck. I guess it would be possible not to include an ACL if it can't be parsed. However, then the series API and the event API will behave differently there. And changing the latter is not something I want to spend time on right now. So maybe this is okay since it falls back to the default ACL, which is "allow nothing". Opinions?

Reply:
> However, then the series API and the event API will behave differently there.

That's a shame. When was that part of the API added? Would it still be possible to change it without breaking backwards compatibility? 

I do think that using the default ACL as fallback is not a good idea. For software using the API, this can easily lead to strange errors down the line. Even if these programs would explicitly check for an empty ACL and interpret it as "ACL cannot be parsed", then we have no way of actually communicating an empty ACL. (Not sure if that can ever happen.) 

In my opinion, if parsing the ACL should never happen (as in: if it fails, there is a bug somewhere else in Opencast), then the API should return 5xx in that case. If parsing failure is somewhat expected and does not imply a bug in Opencast, then the API should explicitly return `acl: null` or something like that to clearly indicate broken ACL. 

In this week alone, strange and unexpected behaviors of the API were brought up twice in our internal meetings. Everyone just sighs, maybe laughs, but then carries on. If the Opencast community just keeps adding to the pile of strange API things, the situation is only gonna get worse. I am obviously not blaming you by the way; it's just the community's stance on this issue that I kind of criticize. 

So I personally would not merge this PR, but fix the behavior as stated above and if somehow possible also fix the behavior of the event API to make them consistent. *That said*, this is not for me to decide. Maybe the benefits of having this API in 9.x outweighs the disadvantages of the strangeness. Maybe this can still be fixed after the feature freeze. And I probably just lack experience in this community to know what's best.

Reply:
I understands your frustration with the API in general and swallowing errors specifically , though I think this case isn't quite as bad. At least this way, Opencast offers a reasonable fallback (deny everything) instead of leaving every external system to figure this out for themselves.
The point of changing behaviour of existing API endpoints is that this can (per our rules) only be done for new API versions, so you would have to add a condition to the events API that handles broken ACLs differently for different versions. This is a general problem which stems from the trade-off between a stable API and improving things. Not sure if this case is worth it.